### ISSUE TITLE
document TRAIT_LOYAL changes

### ISSUE THREAD

**Issue by gfgtdf** [wesnoth/wesnoth#4416]
the changes to TRAIT_LOYAL are not yet documented

**Comment by hexagonrecursion** [comment]
Could you please elaborate? #define TRAIT_LOYAL have remained basically unchanged for about 2 years.

https://github.com/wesnoth/wesnoth/blob/c97d2cb6fb9ff4f8c6ad221b5c7b3922f40a954d/data/core/macros/traits.cfg#L4

**Comment by soliton-** [comment]
I assume this is about https://github.com/wesnoth/wesnoth/commit/5cb9369

**Comment by CelticMinstrel** [comment]
Was this ever documented?

**Comment by Pentarctagon** [comment]
What about the change needs to be documented?

**Comment by CelticMinstrel** [comment]
I'm not sure what I was thinking about when I originally asked that, but looking at the commit, I can see at least two things:

* `IS_HERO`, `IS_LOYAL`, and I think also `IS_EXPENDABLE_LEADER` are now deprecated
* `TRAIT_LOYAL` has a new optional argument (this is probably somehow a job for `wmlscope`)

**Comment by Pentarctagon** [comment]
It doesn't look like any of those first three macros are actually marked as deprecated though?
https://github.com/wesnoth/wesnoth/blob/536d6a226cd4ed76b034ef06a521e56eef309c68/data/core/macros/image-utils.cfg#L18-L55

The [automatically generated macro documentation](url) seems to have a problem with optional arguments though, given its description of `TRAIT_LOYAL` is
> endarg Units with trait Loyal have no upkeep.

**Comment by CelticMinstrel** [comment]
> The automatically generated macro documentation seems to have a problem with optional arguments 

I don't think it understands them at all and just interprets them as comments.

> It doesn't look like any of those first three macros are actually marked as deprecated though?

Well, that was kinda my point. They're not _marked_ as deprecated, but they probably should be, as clearly TRAIT_HERO etc is intended to replace them.

**Comment by Pentarctagon** [comment]
Well, unless they're actually deprecated, documentation can't be updated to say that they're deprecated.  And given the macro documentation itself is automatically generated, is there anything that needs to be done here?

**Comment by CelticMinstrel** [comment]
I'm not sure. There might be some references on the wiki that need updating. Besides that, enabling wmlscope to document optional arguments should probably be a separate issue, and marking these deprecated could be done whenever and maybe doesn't need an issue?

**Comment by Pentarctagon** [comment]
Actually it looks like it already does since b39ca0e5488e0cb16247588b73bec38e20d14178?  I guess the page hasn't been regenerated for a while.

**Comment by CelticMinstrel** [comment]
That commit by itself doesn't seem sufficient, but I suppose there must be other related commits as well that implement it. It should probably have some support for adding doc comments on the optional arguments, but that too could be a separate issue (if it wasn't already done).

**Comment by cooljeanius** [comment]
> I'm not sure what I was thinking about when I originally asked that, but looking at the commit, I can see at least two things:
> 
>     * `IS_HERO`, `IS_LOYAL`, and I think also `IS_EXPENDABLE_LEADER` are now deprecated
> 

`wmllint` should warn about this, IMO

>     * `TRAIT_LOYAL` has a new optional argument (this is probably somehow a job for `wmlscope`)
> 
> 
> There's also the proliferation of new related macros (eg `TRAIT_HERO`), but as long as the generated macro documentation is kept up-to-date, that'd just be a question of searching for references to the deprecated stuff on the wiki and such.

**Comment by cooljeanius** [comment]
btw @Toranks this bug is kinda related to some of your recent ANO changes

**Comment by Toranks** [comment]
> btw @Toranks this bug is kinda related to some of your recent ANO changes

These changes are practically the same
https://github.com/wesnoth/wesnoth/commit/5cb9369

**Comment by Pentarctagon** [comment]
What is there to do here?

**Comment by cooljeanius** [comment]
> What is there to do here?

I think `wmllint` still needs to warn that `{IS_HERO}`, `{IS_LOYAL}`, and possibly also also `{IS_EXPENDABLE_LEADER}` are deprecated, but I (or someone™) will have to double-check to make sure...
